src/Message/MessageHelpers.ts

```diff
@@ -34,21 +34,27 @@ function showMessage(name: string): void {
 }
 
 //Adds a message to a server
-function addMessageToServer(msg: Message, serverHostname: string): void {
-  const server = GetServer(serverHostname);
-  if (server == null) {
-    console.warn(`Could not find server ${serverHostname}`);
+function addMessageToServer(msg: Message): void {
+  //Short-circuit if the message has already been saved
+  if (recvd(msg)) {
     return;
   }
-  for (let i = 0; i < server.messages.length; ++i) {
-    const other = server.messages[i];
-    if (msg.filename === other) {
-      return; //Already exists
-    }
+  const server = GetServer("home");
+  if (server == null) {
+    throw new Error("The home server doesn't exist. You done goofed.");
   }
   server.messages.push(msg.filename);
 }
 
+//Returns whether the given message has already been received
+function recvd(msg: Message): boolean {
```

Comment:
try to use full words just makes it more readable :)

Reply:
Calling it recvd was actually naming parity with the removed msg.recvd flag. I suppose there was no particular reason to keep the name.